src/index.ts

```diff
@@ -16,46 +16,34 @@ if (argv.help || argv.h) {
   process.exit(0);
 }
 
-// Get configuration from environment variables and command line
-const port = process.env.PORT ? parseInt(process.env.PORT) : 3000;
-const fmpToken = argv["fmp-token"] || process.env.FMP_ACCESS_TOKEN;
-
-// Parse toolSets argument from command line or environment variable
-let toolSets: ToolSet[] = [];
-const toolSetsInput =
-  argv["fmp-tool-sets"] || argv["tool-sets"] || argv["toolSets"] || process.env.FMP_TOOL_SETS;
-if (toolSetsInput && typeof toolSetsInput === "string") {
-  toolSets = toolSetsInput.split(",").map((s) => s.trim()) as ToolSet[];
-}
-
-// Parse dynamic tool discovery from command line or environment variable
-const dynamicToolDiscovery =
-  argv["dynamic-tool-discovery"] === true ||
-  argv["dynamicToolDiscovery"] === true ||
-  validateDynamicToolDiscoveryConfig(process.env.DYNAMIC_TOOL_DISCOVERY);
-
-// Validate tool sets
-const availableToolSets = getAvailableToolSets().map(({ key }) => key);
-const invalidToolSets = toolSets.filter(
-  (ts) => !availableToolSets.includes(ts)
-);
-if (invalidToolSets.length > 0) {
-  console.error(`Invalid tool sets: ${invalidToolSets.join(", ")}`);
-  console.error(`Available tool sets: ${availableToolSets.join(", ")}`);
-  process.exit(1);
+function main() {
+  // Initialize the ServerModeEnforcer with env vars and CLI args
+  // This will also validate tool sets and exit if invalid ones are found
+  ServerModeEnforcer.initialize(process.env, argv);
+  
+  const PORT = process.env.PORT ? parseInt(process.env.PORT) : 3000;
+  const fmpToken = argv["fmp-token"] || process.env.FMP_ACCESS_TOKEN;
+
+  const mcpServer = new FmpMcpServer(
+    {
+      accessToken: fmpToken,
+      cacheOptions: {
+        maxSize: 25,
+        ttl: 1000 * 60 * 60 * 2, // 2 hours
+      },
+    }
+  )
+
+  mcpServer.start(PORT);
```

Comment:
This is the recommended way by Smithery? I remember they support the export default approach (but maybe it's runtime dependent) https://smithery.ai/docs/getting_started/quickstart_build

Reply:
I wanted to support it as well since I think glama would need a way to pass port for their healthcheck ping.
unsure yet though.